service/controller/azure/v7/resource_set.go

```diff
@@ -247,6 +232,21 @@ func NewResourceSet(config ResourceSetConfig) (*controller.ResourceSet, error) {
 		}
 	}
 
+	var guestClusterService guestcluster.Interface
```

Comment:
I find it confusing that we now have both `guestClusterService` and `tenantClusterService`, not sure what we can do about this though.